refaster-runner/src/main/java/tech/picnic/errorprone/refaster/runner/RefasterRuleSelector.java

```diff
@@ -0,0 +1,554 @@
+      case REMAINDER:
+        return "%";
+      case PLUS:
+        return "+";
+      case MINUS:
+        return "-";
+      case LEFT_SHIFT:
+        return "<<";
+      case RIGHT_SHIFT:
+        return ">>";
+      case UNSIGNED_RIGHT_SHIFT:
+        return ">>>";
+      case LESS_THAN:
+        return "<";
+      case GREATER_THAN:
+        return ">";
+      case LESS_THAN_EQUAL:
+        return "<=";
+      case GREATER_THAN_EQUAL:
+        return ">=";
+      case EQUAL_TO:
+        return "==";
+      case NOT_EQUAL_TO:
+        return "!=";
+      case AND:
+        return "&";
+      case XOR:
+        return "^";
+      case OR:
+        return "|";
+      case CONDITIONAL_AND:
+        return "&&";
+      case CONDITIONAL_OR:
+        return "||";
+      case MULTIPLY_ASSIGNMENT:
+        return "*=";
+      case DIVIDE_ASSIGNMENT:
+        return "/=";
+      case REMAINDER_ASSIGNMENT:
+        return "%=";
+      case PLUS_ASSIGNMENT:
+        return "+=";
+      case MINUS_ASSIGNMENT:
+        return "-=";
+      case LEFT_SHIFT_ASSIGNMENT:
+        return "<<=";
+      case RIGHT_SHIFT_ASSIGNMENT:
+        return ">>=";
+      case UNSIGNED_RIGHT_SHIFT_ASSIGNMENT:
+        return ">>>=";
+      case AND_ASSIGNMENT:
+        return "&=";
+      case XOR_ASSIGNMENT:
+        return "^=";
+      case OR_ASSIGNMENT:
+        return "|=";
+      default:
+        throw new IllegalStateException("Cannot convert Tree.Kind to a String: " + kind);
+    }
+  }
+
+  private static final class RefasterIntrospection {
+    private static final String UCLASS_IDENT_FQCN = "com.google.errorprone.refaster.UClassIdent";
+    // XXX: Probably there is a better way to fix this... For a few BeforeTemplates like
+    // `ImmutableMapBuilder` the algorithm wouldn't match so created this fix for now. About 10
+    // templates would always match.
+    private static final String AUTO_VALUE_UCLASS_IDENT_FQCN =
+        "com.google.errorprone.refaster.AutoValue_UClassIdent";
+    private static final Class<?> UCLASS_IDENT = getClass(UCLASS_IDENT_FQCN);
+    private static final Class<?> UCLASS_AUTOVALUE_IDENT = getClass(AUTO_VALUE_UCLASS_IDENT_FQCN);
+    private static final Method METHOD_REFASTER_RULE_BEFORE_TEMPLATES =
+        getMethod(RefasterRule.class, "beforeTemplates");
+    private static final Method METHOD_EXPRESSION_TEMPLATE_EXPRESSION =
+        getMethod(ExpressionTemplate.class, "expression");
+    private static final Method METHOD_BLOCK_TEMPLATE_TEMPLATE_STATEMENTS =
+        getMethod(BlockTemplate.class, "templateStatements");
+    private static final Method METHOD_USTATIC_IDENT_CLASS_IDENT =
+        getMethod(UStaticIdent.class, "classIdent");
+    private static final Method METHOD_UCLASS_IDENT_GET_TOP_LEVEL_CLASS =
+        getMethod(UCLASS_IDENT, "getTopLevelClass");
+    private static final Method METHOD_UANY_OF_EXPRESSIONS = getMethod(UAnyOf.class, "expressions");
+
+    static boolean isUClassIdent(IdentifierTree tree) {
```

Comment:
Let's use the full word.
```suggestion
    static boolean isUClassIdentifier(IdentifierTree tree) {
```